napari/_qt/dialogs/qt_plugin_dialog.py

```diff
@@ -64,27 +98,46 @@ def __init__(
         enabled: bool = True,
         installed: bool = False,
         npe_version=1,
+        versions_conda: list = [],
+        versions_pypi: list = [],
     ):
         super().__init__(parent)
+        self.url = url
+        self._versions = {}
+        self._versions['Conda'] = versions_conda
+        self._versions['PyPI'] = versions_pypi
```

Comment:
Could we change it to two variables, Use global variables to store string values, or annotate as TypedDict? 

I see here some form of magic constraints that should not be used.